packages/@aws-cdk/aws-cloudfront/lib/edge-function.ts

```diff
@@ -0,0 +1,271 @@
+  }
+  public metricInvocations(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricInvocations({ ...props, region: EdgeFunctionExperimental.EDGE_REGION });
+  }
+  public metricThrottles(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.lambda.metricThrottles({ ...props, region: EdgeFunctionExperimental.EDGE_REGION });
+  }
+  public addEventSource(source: lambda.IEventSource): void {
+    return this.lambda.addEventSource(source);
+  }
+  public configureAsyncInvoke(options: lambda.EventInvokeConfigOptions): void {
+    return this.lambda.configureAsyncInvoke(options);
+  }
+
+  /** Create a function in-region */
+  private createInRegionFunction(id: string, props: lambda.FunctionProps): FunctionConfig {
+    const role = props.role ?? defaultLambdaRole(this, id);
+    const edgeFunction = new lambda.Function(this, 'Fn', {
+      ...props,
+      role,
+    });
+
+    return { edgeFunction, edgeArn: edgeFunction.currentVersion.edgeArn, functionStack: this.stack };
+  }
+
+  /** Create a support stack and function in us-east-1, and a SSM reader in-region */
+  private createCrossRegionFunction(id: string, props: lambda.FunctionProps): FunctionConfig {
+    const parameterNamePrefix = 'EdgeFunctionArn';
+    const parameterName = `${parameterNamePrefix}${id}`;
+    const functionStack = this.edgeStack();
+    this.stack.addDependency(functionStack);
+
+    const edgeFunction = functionStack.addEdgeFunction(id, parameterName, props);
+    // This is used to determine when the function has changed, to refresh the ARN from the custom resource.
+    const edgeFunctionHash = calculateFunctionHash(edgeFunction);
+
+    // Prefix of the parameter ARN that applies to all EdgeFunctions.
+    // This is necessary because the `CustomResourceProvider` is a singleton, and the `policyStatement`
+    // must work for multiple EdgeFunctions.
+    const parameterArnPrefix = this.stack.formatArn({
+      service: 'ssm',
+      region: EdgeFunctionExperimental.EDGE_REGION,
+      resource: 'parameter',
+      resourceName: parameterNamePrefix + '*',
+    });
+
+    const resourceType = 'Custom::CrossRegionStringParameterReader';
+    const serviceToken = CustomResourceProvider.getOrCreate(this, resourceType, {
+      codeDirectory: path.join(__dirname, 'edge-function'),
+      runtime: CustomResourceProviderRuntime.NODEJS_12,
+      policyStatements: [{
+        Effect: 'Allow',
+        Resource: parameterArnPrefix,
+        Action: ['ssm:GetParameter'],
+      }],
+    });
+    const resource = new CustomResource(this, 'ArnReader', {
+      resourceType: resourceType,
+      serviceToken,
+      properties: {
+        Region: EdgeFunctionExperimental.EDGE_REGION,
+        ParameterName: parameterName,
+        RefreshToken: edgeFunctionHash,
+      },
+    });
+    const edgeArn = resource.getAttString('FunctionArn');
```

Comment:
Can we use an automatically assigned physical name instead of an SSM param/custom resource here?

Reply:
Got it. Thanks for the explanation. Maybe worth documenting this decision process in the code shortly